### PR TITLE
knn_model

### PR DESCRIPTION
**Pull request by Sar-thak-3**
Added python knn model at src/slowmokit/models/knn.hpp

### CODE REVIEW

**Comment by UtkarshGoelUT** [comment]
Hey 
Please refer to the linear regression model to see the best practices and try to implement those. 
```
https://github.com/PEC-CSS/slowmokit/tree/main/src/slowmokit/methods/linear_model/linear_regression
```
Also, could you try adding a single model per PR?

**Comment by UtkarshGoelUT** [comment]
Also for each of the models you add, kindly add example for the same. Refer to linear regression model. 
```
https://github.com/PEC-CSS/slowmokit/blob/main/examples/linear_models/linear_regression.cpp
```

File: src/slowmokit/models/Kmeans.hpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
+#include "core.hpp"
+using namespace std;
```
Comment (ken1000minus7) [inline src/slowmokit/models/Kmeans.hpp:2]: Do not use `using namespace std` anywhere, use `std::` wherever applicable instead

File: src/slowmokit/models/bernoulliNB.hpp [hunk 2]
```diff
@@ -0,0 +1,59 @@
+#include "core.hpp"
+using namespace std;
+
+class BernoulliNB{
+    double prior_prob(vector<int> y_train,int label){
```
Comment (ken1000minus7) [inline src/slowmokit/models/bernoulliNB.hpp:5]: According to our guidelines variables and function names should be in camel case **not** snake case

**Review by ken1000minus7 — CHANGES_REQUESTED** [review]

**Review by harshjohar — CHANGES_REQUESTED** [review]
## Following changes must be done before PR
- Fix the directory structure according to the mentioned conventions in [README.md](../../README.md)
- Use a IDE like CLion, to automatically configure CMake, otherwise, configure it yourself
- use `hpp` files only to declare functions, all the definitions must be done in the `cpp` file.
- The following is the correct directory structure for a new model

> DO NOT USE `use namespace std;`

```
src
  -slowmokit
    - methods
      - <Model type>
         - <Model Name>/
         - <Model Name>.hpp
```

**Review by uttammittal02 — CHANGES_REQUESTED** [review]
You didn't make use of the variable 'k' in the code of kNN passed as a parameter in the function predict.
Review Line 25 of this file

**Comment by UtkarshGoelUT** [comment]
> Can you please guide me how do I now, test the working of all models prepared according to the structure defined in README.md , as I never worked with Clion?

Create a file in the examples folder, and run that file.
Refer to the [linear regression example](https://github.com/PEC-CSS/slowmokit/blob/main/examples/linear_models/linear_regression.cpp) for the same.

**Pull request closed without merging**